examples/module1/lesson5/app-performance/components/Comments.tsx

```diff
@@ -0,0 +1,85 @@
+import { Comment } from '../types/Comment.ts';
+import axios from 'axios';
+import { useState } from 'react';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+export const createArticle = async (comment: Comment) => {
+  const response = await axios.post('http://localhost:3001/api/data/comments?timeout=5000', comment);
+  return response.data;
+};
+
+export const Comments = ({ data }: { data: any }) => {
+
+  const [newComment, setNewComment] = useState('');
+  const [newRating, setNewRating] = useState('');
+
+  const queryClient = useQueryClient();
+
+  const { isPending, variables, mutate } = useMutation({
+    mutationFn: createArticle,
+    onSettled: async () => {
+      return await queryClient.invalidateQueries({ queryKey: ['comments'] });
+    }
+  });
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
brakuje mi tutaj case'a gdy mutacja zwróci error. Jeden to np error podczas walidacji i przywrócenie wartości do formularza aby user poprawił form. Drugi case to inny błąd i możliwość np. refetcha

Reply:
Racja. W sumie zrobiłem to po linii najmniejszego oporu, by tylko mieć ogarnięty optimistic update. Poprawię :-)